mk/config.mk

```diff
@@ -1158,3 +1158,20 @@ $(eval $(call cfg-depends-all,CFG_WIDEVINE_HUK,CFG_DT))
 # DT node "/options/op-tee/widevine" to some specific TAs.
 CFG_WIDEVINE_PTA ?= n
 $(eval $(call cfg-depends-all,CFG_WIDEVINE_PTA,CFG_DT CFG_WIDEVINE_HUK))
+
+# Semihosting is a debugging mechanism that enables code running on an embedded
+# system (also called the target) to communicate with and use the I/O of the
+# host computer.
+CFG_SEMIHOSTING ?= n
+
+# Enable the semihosting console driver which inputs/outputs the characters
+# from/to the file or debug terminal on the semihosting host computer.
+#   - Specify CFG_SEMIHOSTING_CONSOLE_FILE="{your_log_file}" to output the
+#     characters to that file. Output to "optee.log" by default.
+#   - Specify CFG_SEMIHOSTING_CONSOLE_FILE=NULL to output the characters to the
+#     semihosting host terminal.
+CFG_SEMIHOSTING_CONSOLE ?= n
+ifeq ($(CFG_SEMIHOSTING_CONSOLE),y)
+$(call force,CFG_SEMIHOSTING,y)
+endif
+CFG_SEMIHOSTING_CONSOLE_FILE ?= "optee.log"
```

Comment:
to prevent a build error if one sets CFG_SEMIHOSTING_CONSOLE_FILE to an empty value, I suggest to add 
```c
ifeq ($(CFG_SEMIHOSTING_CONSOLE_FILE),)
$(error CFG_SEMIHOSTING_CONSOLE_FILE cannot by empty)
endif
```
It will fault on no-value case as well as on empty string case.